core/tests/behavior/utils.rs

```diff
@@ -110,255 +103,89 @@ macro_rules! blocking_trials {
-        }
+        self.new_file_with_range(uuid::Uuid::new_v4().to_string(), 1..max_size)
     }
-}
-
-impl ObjectWriterFuzzer {
-    /// Create a new fuzzer.
-    pub fn new(name: &str, size: Option<usize>) -> Self {
-        Self {
-            name: name.to_string(),
-            bs: Vec::new(),
-
-            size,
-            cur: 0,
 
-            rng: thread_rng(),
-            actions: vec![],
-        }
+    pub fn new_file_with_path(
+        &self,
+        op: impl Into<Operator>,
+        path: &str,
+    ) -> (String, Vec<u8>, usize) {
+        let max_size = op
+            .into()
+            .info()
+            .full_capability()
+            .write_total_max_size
+            .unwrap_or(4 * 1024 * 1024);
+
+        self.new_file_with_range(path, 1..max_size)
     }
 
-    /// Generate a new action.
-    pub fn fuzz(&mut self) -> ObjectWriterAction {
-        let max = if let Some(size) = self.size {
-            size - self.cur
-        } else {
-            // Set max to 1MiB
-            1024 * 1024
-        };
-
-        let size = self.rng.gen_range(0..max);
-
-        let mut bs = vec![0; size];
-        self.rng.fill_bytes(&mut bs);
+    /// Create a new file with random content in range.
+    fn new_file_with_range(
+        &self,
+        path: impl Into<String>,
+        range: impl SampleRange<usize>,
+    ) -> (String, Vec<u8>, usize) {
+        let path = path.into();
+        self.paths.lock().unwrap().push(path.clone());
 
-        let bs = Bytes::from(bs);
-        self.bs.extend_from_slice(&bs);
-        self.cur += bs.len();
+        let mut rng = thread_rng();
 
-        let action = ObjectWriterAction::Write(bs);
-        debug!("{} perform fuzz action: {:?}", self.name, action);
+        let size = rng.gen_range(range);
+        let mut content = vec![0; size];
+        rng.fill_bytes(&mut content);
 
-        self.actions.push(action.clone());
-
-        action
+        (path, content, size)
     }
 
-    /// Check if read operation is expected.
-    pub fn check(&mut self, actual_bs: &[u8]) {
-        assert_eq!(
-            self.bs.len(),
-            actual_bs.len(),
-            "check failed: expected len is different with actual len, actions: {:?}",
-            self.actions
-        );
-
-        assert_eq!(
-            format!("{:x}", Sha256::digest(&self.bs)),
-            format!("{:x}", Sha256::digest(actual_bs)),
-            "check failed: expected bs is different with actual bs, actions: {:?}",
-            self.actions,
-        );
+    /// Perform cleanup
+    pub async fn cleanup(&self, op: impl Into<Operator>) {
+        let op = op.into();
+        let paths: Vec<_> = mem::take(self.paths.lock().unwrap().as_mut());
+        for path in paths.iter() {
+            // We try our best to cleanup fixtures, but won't panic if failed.
+            let _ = op.delete(path).await.map_err(|err| {
+                log::error!("fixture cleanup path {path} failed: {:?}", err);
+            });
```

Comment:
May we use `remove_all` here?

Reply:
We can change to `remove` if the speed is too slow.

Reply:
It's not a matter of being slow. It may not clean up completely. 
For example, if there is a nesting path.

Reply:
> It's not a matter of being slow. It may not clean up completely. For example, if there is a nesting path.

While it may be true, this PR isn't aiming for a perfect solution; we've simply made some improvements to the existing codebase.

The `remove_all` function is avoided as it may impact other active services in the event of a bug.